app/controllers/uploads_controller.rb

```diff
@@ -462,6 +482,22 @@ def complete_multipart
 
   protected
 
+  def multipart_store(upload_type)
+    if upload_type == "backup"
+      check_backup_permission
+      BackupRestore::BackupStore.create
+    else
+      Discourse.store
+    end
+  end
+
+  def check_backup_permission
```

Comment:
This is some of the jankyness mentioned in the PR description that I hope to address in a subsequent PR.